README.md

```diff
@@ -0,0 +1,400 @@
+    RTPS_DllAPI DomainParticipant(
+            DomainId_t domain_id,
+            const DomainParticipantQos& qos,
+            DomainParticipantListener* listener,
+            const StatusMask& mask = StatusMask::all());
+};
+```
+
+### Constructor Initializer List
+
+Constructor initializer parameters should be each in one line.
+Except the first parameter which is preceded by a colon `:`, the rest should have the comma `,` before the name.
+There should also be a space between the colon and the comma following the [horizontal whitespace rules](#horizontal-whitespace).
+
+The indentation in this case, instead of being +8 spaces, is only of +4 spaces.
+
+```c++
+DynamicType::DynamicType()
+    : descriptor_(nullptr)
+    , name_("")
+    , kind_(TK_NONE)
+    , is_key_defined_(false)
+{
+}
+```
+
+### Namespace Formatting
+
+When declaring a namespace, the open brace should be in the same line as the declaration, separated one space from the namespace name.
+This is the only exception in the code style where the opening brace is not in its own line.
+
+Namespaces do not add an extra level of indentation.
+
+There should be a comment after the closing brace of the namespace, even more if the scope of the namespace is great enough and there may be doubts about which namespace is being closed.
+
+```c++
+namespace network {
+namespace ip {
+
+class IPv4 // Correct. No extra indentation in namespaces
+{
+public:
+
+    long socket;
+};
+
+} // ip
+} // network
+```
+
+### Templates
+
+[//]: # (The uncrustify configuration file does not enforce that the template parameters should be each in a different line.)
+
+Template declaration should be in a different line from the entity it modifies.
+
+```c++
+template<typename K, typename V>
+struct map_size_helper : public detail::pool_size_helper<detail::map_node_size<K, V>::value>
+{
+};
+```
+
+### Preprocessor Directives
+
+The hash mark `#` that starts a preprocessor directive should always be at the beginning of the line.
+Even when preprocessor directives are within the body of indented code, the directives should start at the beginning of the line.
+
+There should be a comment after `#endif` indicating the `#if` which is ending.
+This is even more important when the scope of the `#if` is great enough and there may be doubts.
+
+```c++
+    ProtocolVersion_t():
+#if HAVE_SECURITY
+        // As imposed by DDSSEC11-93
+        ProtocolVersion_t(2,3)
+#else
+        ProtocolVersion_t(2,2)
+#endif // HAVE_SECURITY
+```
+
+There must be spaces around preprocessor `##` concatenation operator.
+
+```c++
+#define PASTE2( x, y) x ## y
+```
+
+## Struct vs Classes
+
+Use a struct only for passive objects that carry data; everything else is a class.
+It is advisable that structs do not provide any method, although public access methods could be accepted.
+However, as the struct members should be public, there would be no need to provide those methods.
```

Comment:
I'd be more strict and request all data in the struct to be public. The only case I can think of for non-public data is if the struct needs to keep an internal state, and in that case, it should be a class.
I don't know if uncrustify can enforce this, but we still should be able to state it in the style guide.

Reply:
Done.